### PR TITLE
TablePanel: Add solid  background cell display mode

### LINKED ISSUE #29074: feature-request: toggle gradient/plain color in table cell background
since grafana 7.0 there is by default a gradient in the background of a colored table cell. 

Sometimes this does not look straight :-)

So I suggest a new option for table cells: toggle plain color/gradient

### PR DESCRIPTION
**Pull request by dprokop**
Closes https://github.com/grafana/grafana/issues/29074

I was considering adding a conditional boolean switch when a display mode is set to Color Background, but I think this is more readable and discoverable.

### CODE REVIEW

**Comment by torkelo** [comment]
Wonder if we should wait with introducing a new mode make this mode sub-option (https://github.com/grafana/grafana/issues/31315) 

Then background color could mode could have a sub-option 
`Background look`  radio: Solid | Gradient |  Badge 

The badge look would not color the full cell but just the value with some round padding:

File: docs/sources/panels/visualizations/table/table-field-options.md [hunk 1]
```diff
@@ -42,9 +42,9 @@ If thresholds are set, then the field text is displayed in the appropriate thres
 
 {{< docs-imagebox img="/img/docs/tables/color-text.png" max-width="500px" caption="Color text" class="docs-image--no-shadow" >}}
 
-### Color background
+### Color background (gradient or solid)
 
-If thresholds are set, then the field background is displayed in the appropriate threshold color.
+If thresholds are set, then the field background is displayed in the appropriate threshold color.
```
Comment (achatterjee-grafana) [inline docs/sources/panels/visualizations/table/table-field-options.md:47]:
```suggestion
If thresholds are set, then the field background displays in the appropriate threshold color. 
```

**Review by achatterjee-grafana — COMMENTED** [review]
Minor change request.

File: docs/sources/panels/visualizations/table/table-field-options.md (hunk 1 above)
Comment (osg-grafana) [reply docs/sources/panels/visualizations/table/table-field-options.md:47]: +1

File: docs/sources/panels/visualizations/table/table-field-options.md [hunk 2]
```diff
@@ -44,7 +44,7 @@ If thresholds are set, then the field text is displayed in the appropriate thres
 
 ### Color background (gradient or solid)
 
-If thresholds are set, then the field background is displayed in the appropriate threshold color. 
+If thresholds are set, then the field background displays in the appropriate threshold color.
```
Comment (oddlittlebird) [inline docs/sources/panels/visualizations/table/table-field-options.md:47]: "Displays" is a transitive verb. It needs a direct object or to be in the passive.

**Review by oddlittlebird — DISMISSED** [review]

**Comment by oddlittlebird** [comment]
@dprokop, please add a milestone

**Comment by dprokop** [comment]
> Then background color could mode could have a sub-option
> Background look radio: Solid | Gradient | Badge

@torkelo I was thinking straight away about adding a conditional option for this, that would UI-wise be close to what you are suggesting here. But when I tried it didn't feel right, the conditional options add unnecessary complexity IMO. 

I do agree with you about the refactor into an object, but don't think we are going to do that very soon, unfortunately. Having migration from this new mode I'm adding shouldn't be much of a pain (we gonna have to have a migration anyway), so I would probably proceed with this one if you don't strongly disagree.

File: docs/sources/panels/visualizations/table/table-field-options.md (hunk 2 above)
Comment (dprokop) [inline docs/sources/panels/visualizations/table/table-field-options.md:47]:
```suggestion
If thresholds are set, then the field background is displayed in the appropriate threshold color. 
```

Comment (osg-grafana) [reply docs/sources/panels/visualizations/table/table-field-options.md]: Then leave it as `is displayed`, where the doer is unimportant.

**Comment by achatterjee-grafana** [comment]
The milestone is v8.0. Are we checking in docs are the v8.0 release already?

**Comment by osg-grafana** [comment]
> The milestone is v8.0. Are we checking in docs are the v8.0 release already?

If not, where else should we be developing for three months?

**Comment by achatterjee-grafana** [comment]
@osg-grafana, we reverted some v8.0 doc changes just last week, so I am confused about the timeline!

**Comment by dprokop** [comment]
@achatterjee-grafana @osg-grafana Not sure I understand what you mean? We have plenty of v8 milestoned pull requests already. Do the docs changes affect your workflow somehow?

**Comment by osg-grafana** [comment]
> @achatterjee-grafana @osg-grafana Not sure I understand what you mean? We have plenty of v8 milestoned pull requests already. Do the docs changes affect your workflow somehow?

@achatterjee-grafana is leading the effort to get docs versioned, so that is why I am asking how we can contribute to v8.0 docs now in source code rather than wait for all changes to go in in the form of some tight bottleneck before the release.

**Comment by osg-grafana** [comment]
> @osg-grafana, we reverted some v8.0 doc changes just last week, so I am confused about the timeline!

Where did they get reverted to?

**Comment by dprokop** [comment]
> @achatterjee-grafana is leading the effort to get docs versioned, so that is why I am asking how we can contribute to v8.0 docs now in (technical documentation) source code rather than wait for all changes to go in in the form of some tight bottleneck before the release

@osg-grafana I don't think this PR is the best place to discuss that. Let's talk about it on slack and we need to sync between different teams too.

**Comment by achatterjee-grafana** [comment]
@oddlittlebird, can you chime in when you have a moment. I would like us to be consistent with how we handle doc changes for v8.0. Keep in mind https://github.com/grafana/grafana/pull/32335 from Ryan, that we requested. Thanks!

**Comment by achatterjee-grafana** [comment]
> @achatterjee-grafana @osg-grafana Not sure I understand what you mean? We have plenty of v8 milestoned pull requests already. Do the docs changes affect your workflow somehow?

Let me get back to you on this. Thanks :)

**Comment by oddlittlebird** [comment]
@dprokop, right now, all doc changes that are merged to master are also published to the doc site. 

If the change is something that can be easily labeled as unavailable until Grafana 8, then we can push it.

If they change current content in a way that would be confusing, then we need to capture it somewhere else.

@robbymilo and @zuchka and @torkelo, this is relevant to our discussion on #27384.

**Review by torkelo — APPROVED** [review]

**Pull request merged**